### PR TITLE
AC-700 Refined custom dialog box in add patient activity

### PR DESCRIPTION
**Pull request by rishabh-997**
## Description of what I changed
1. Removed previous custom dialog fragment
2. Created Another dialog fragment in which we can pass more parameters, has a recycler view to increasing its functionality and readability.
3. Created interface to handle click events between both fragments.

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-700

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [x] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CameraOrGalleryPickerDialog.java [hunk 1]
```diff
@@ -14,89 +14,122 @@
...
     }
 
-    @NonNull
+    @Nullable
     @Override
-    public Dialog onCreateDialog(Bundle savedInstanceState) {
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.dialog_my_custom, container, false);
+        RecyclerView recyclerView = view.findViewById(R.id.recyclerView_dialog);
+        list.add(new CustomDialogModel("Take a photo", R.drawable.ic_photo_camera));
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CameraOrGalleryPickerDialog.java:52]: Put `"Take a photo"` in a string resource.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CameraOrGalleryPickerDialog.java [hunk 2]
```diff
@@ -14,89 +14,122 @@
...
+        list.add(new CustomDialogModel("Choose photo", R.drawable.ic_photo_library));
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CameraOrGalleryPickerDialog.java:53]: Put `"Choose photo"` in a string resource.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CameraOrGalleryPickerDialog.java [hunk 3]
```diff
@@ -14,89 +14,122 @@
...
+        if (showRemoveButton)
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CameraOrGalleryPickerDialog.java:54]: See the [OpenMRS Java conventions](https://wiki.openmrs.org/display/docs/Java+Conventions), if statements should always have curly braces.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CameraOrGalleryPickerDialog.java [hunk 4]
```diff
@@ -14,89 +14,122 @@
...
+            list.add(new CustomDialogModel("Remove photo", R.drawable.ic_photo_delete));
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/dialog/CameraOrGalleryPickerDialog.java:55]: Place `"Remove photo"` in a string resource.

File: openmrs-client/src/main/res/layout/list_gallery_or_camera_item.xml [hunk 5]
```diff
@@ -13,28 +12,32 @@
...
 
     <TextView
         android:id="@+id/textView"
-        android:layout_width="wrap_content"
+        android:layout_width="match_parent"
         android:layout_height="wrap_content"
+        android:layout_marginEnd="50dp"
+        android:layout_marginRight="50dp"
+        android:paddingEnd="15dp"
+        android:paddingRight="15dp"
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/layout/list_gallery_or_camera_item.xml:39]: Should be even sized, e.g. `16dp`

**Comment by rishabh-997** [comment]
Done with the changes @f4ww4z sir

**Pull request merged**